### PR TITLE
add categories controller and category model/migration. add counter to s...

### PR DESCRIPTION
**Pull request by angelabier1**
...eeds.rb file. move listings with counter as parameter into the loop. add i += 100 to the end of the loop so that the offset will increase by 100 to get additional listings. add i as an argument for shops, images, trending as well as categories.

### CODE REVIEW

File: rails/etsy_product_catalog/app/views/listings/show.html.erb [hunk 1]
```diff
@@ -1,12 +1,13 @@
 <h1>Product details</h1>
 
 
-    <h2><%= @listing.title %></h2>
-    <p><%= @listing.description %></p>
+<h2><%= @listing.title %></h2>
+<p><%= @listing.description %></p>
+<%= image_tag("#{@listing.main_image}") %>
```
Comment (bgates) [inline rails/etsy_product_catalog/app/views/listings/show.html.erb]: minor point - assuming @listing.main_image is a string, interpolating it is unnecessary. Simply image_tag(@listing.main_image) will work.

File: rails/etsy_product_catalog/app/views/listings/show.html.erb [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
 
 
-    <div class="to_the_right">
-      <h3>Price: $<%= @listing.price %></h3>
-      <%= link_to 'Back', root_path, class: 'back_button' %>
-    </div>
+<div class="to_the_right">
+  <h3>Price: $<%= @listing.price %></h3>
```
Comment (bgates) [inline rails/etsy_product_catalog/app/views/listings/show.html.erb]: Rails has a view helper to handle currency.

File: rails/etsy_product_catalog/config/environments/development.rb [hunk 3]
```diff
@@ -4,7 +4,8 @@
   # In the development environment your application's code is reloaded on
   # every request. This slows down response time but is perfect for development
   # since you don't have to restart the web server when you make code changes.
-  config.cache_classes = false
+
+	config.cache_classes = false
```
Comment (bgates) [inline rails/etsy_product_catalog/config/environments/development.rb:8]: ?

File: rails/etsy_product_catalog/db/migrate/20150301225625_create_categories.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+			t.string :name
+			t.string :category_name
+			t.string :short_name
+			t.string :long_name
+			t.integer :num_children
+
+			t.timestamps null: false
+		end
+	end
+end
```
Comment (bgates) [inline rails/etsy_product_catalog/db/migrate/20150301225625_create_categories.rb:13]: Indentation should be 2 spaces

**Comment by bgates** [comment]
Besides the minor points in my line notes, isn't the whole categories concept somebody else's responsibility?

File: rails/etsy_product_catalog/db/migrate/20150301225625_create_categories.rb (hunk 4 above)
Comment (angelabier1) [inline rails/etsy_product_catalog/db/migrate/20150301225625_create_categories.rb:13]: Okay, I will edit this. I think rubymine has some defaults set up for continuous indentation that I need to change.

**Comment by angelabier1** [comment]
@bgates yes. I was walking through the process with him and a few steps in, I made changes to the seeds.rb file and went ahead and just pushed all the changes. Everything after the initial creation of the controller and model, he is pushing up. Not sure why the changes for the listing views were a part of my changes. I didn't touch those files at all.

File: rails/etsy_product_catalog/config/environments/development.rb (hunk 3 above)
Comment (angelabier1) [inline rails/etsy_product_catalog/config/environments/development.rb:8]: @bgates your guess is as good as mine...

**Pull request closed without merging**